### PR TITLE
Move Siren devices from Switch to Siren entity

### PR DESCRIPTION
**Pull request by iMicknl**
Related:
https://github.com/iMicknl/ha-tahoma/issues/25
https://github.com/iMicknl/ha-tahoma/issues/57
https://github.com/iMicknl/ha-tahoma/issues/453

In a future PR:
- [x] Support setting volume (io:MemorizedSimpleVolumeState)
- Support multiple tones

<a href="https://gitpod.io/#https://github.com/iMicknl/ha-tahoma/pull/538"></a>

### CODE REVIEW

File: custom_components/tahoma/siren.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        """Get whether the siren is in on state."""
+        return self.executor.select_state(CORE_ON_OFF_STATE) == STATE_ON
+
+    async def async_turn_on(self, **kwargs):
+        """Send the on command."""
+
+        if kwargs.get(ATTR_DURATION):
+            duration = kwargs.get(ATTR_DURATION)
+        else:
+            duration = 2 * 60  # 2 minutes
```
Comment (tetienne) [inline custom_components/tahoma/siren.py:56]: Why this default value?

File: custom_components/tahoma/siren.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+        duration_in_ms = duration * 1000
+
+        await self.executor.async_execute_command(
+            COMMAND_RING_WITH_SINGLE_SIMPLE_SEQUENCE,  # https://www.tahomalink.com/enduser-mobile-web/steer-html5-client/vendor/somfy/io/siren/const.js
+            duration_in_ms,  # 2 minutes
```
Comment (tetienne) [inline custom_components/tahoma/siren.py:62]: That's not always 2 min.

File: custom_components/tahoma/siren.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        #     COMMAND_STANDARD,
+        # )
+
+        await self.async_cancel_or_stop_siren(COMMAND_RING_WITH_SINGLE_SIMPLE_SEQUENCE)
+
+    async def async_cancel_or_stop_siren(self, cancel_commands) -> None:
+        """Cancel running execution or send stop command."""
+        # Cancelling a running execution will stop the siren movement
+        # Retrieve executions initiated via Home Assistant from Data Update Coordinator queue
+        exec_id = next(
```
Comment (tetienne) [inline custom_components/tahoma/siren.py:84]: Would be nice to put this in a common place to avoid to duplicate.

Comment (iMicknl) [reply custom_components/tahoma/siren.py:84]: I had the same thought, if an user confirms that this works, I will move this to our executor class.

File: custom_components/tahoma/siren.py (hunk 1 above)
Comment (iMicknl) [reply custom_components/tahoma/siren.py:56]: Took it from here:     https://www.tahomalink.com/enduser-mobile-web/steer-html5-client/vendor/somfy/io/siren/const.js

Comment (tetienne) [reply custom_components/tahoma/siren.py:56]: I mean ATTR_DURATION can be missing?

Comment (iMicknl) [reply custom_components/tahoma/siren.py:56]: Yes, it should be optional, but I would have to double check! :-)

**Review by tetienne — APPROVED** [review]

**Pull request merged**